.github/workflows/validate-pr.yml

```diff
@@ -40,31 +40,33 @@ jobs:
         run: |
           orgflow env:list --output=json | jq . -c | ( read output; echo "::set-output name=environments::$output"; )
 
-      # Find the environment the source branch belongs to:
+      # Find the environment that the source branch belongs to:
       - name: Find source environment
         id: find-source-environment
         run: |
-          echo '${{ steps.env-list.outputs.environments }}' |
-          jq '[.[] | select(.git.branch == "${{ github.head_ref }}") | .name] | select(. | length > 0)[0]' -r | 
-          ( read output; echo "::set-output name=environment-name::$output"; )
-
-      # ... and same for the target branch:
+          set -e          
+          output=`echo '${{ steps.env-list.outputs.environments }}' | jq '[.[] | select(.git.branch == "${{ github.head_ref }}") | .name] | select(. | length > 0)[0]' -r` 
+          echo "::set-output name=environment-name::$output";
+          
+      # ... and then again for the target branch:
       - name: Find target environment
         id: find-target-environment
         run: |
-          echo '${{ steps.env-list.outputs.environments }}' |
-          jq '[.[] | select(.git.branch == "${{ github.base_ref }}") | .name] | select(. | length > 0)[0]' -r | 
-          ( read output; echo "::set-output name=environment-name::$output"; )
+          set -e          
+          output=`echo '${{ steps.env-list.outputs.environments }}' | jq '[.[] | select(.git.branch == "${{ github.base_ref }}") | .name] | select(. | length > 0)[0]' -r` 
+          echo "::set-output name=environment-name::$output";
+
+      # Print a message if at least one branch does not correspond to an environment:

```

Comment:
Would be good to expand this comment a bit to explain when this can happen, and why it should result in a success (like you explained to me in our call).